app/src/test/java/org/mozilla/vrbrowser/GleanMetricsServiceTest.kt

```diff
@@ -0,0 +1,34 @@
+package org.mozilla.vrbrowser
+
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.service.glean.Glean.initialize
+import mozilla.components.service.glean.Glean.setUploadEnabled
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.config.Configuration.Companion.DEFAULT_TELEMETRY_ENDPOINT
+import mozilla.components.service.glean.testing.GleanTestRule
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.vrbrowser.GleanMetrics.*
+
+
+@get:Rule
+val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
+
+class GleanMetricsServiceTest {
+
+    @Test
+    fun testMetricsGathering() {
+        setUploadEnabled(true)
+        val config = Configuration(DEFAULT_TELEMETRY_ENDPOINT, BuildConfig.BUILD_TYPE)
+        initialize(ApplicationProvider.getApplicationContext(), config)
```

Comment:
If I add Glean.initialize(). It will fail as below
```
org.mozilla.vrbrowser.GleanMetricsServiceTest > testMetricsGathering FAILED
    java.lang.IllegalStateException at GleanMetricsServiceTest.kt:24

2 tests completed, 1 failed
```

Reply:
The `GleanTestRule` takes care of the initialization on its own: you should not initialize Glean yourself in tests.